server/src/main/scala/services/graphstages/StaffingStage.scala

```diff
@@ -6,111 +6,116 @@ import akka.stream._
-          runSimulationAndPush()
+          grabAllAvailable()
+          runSimulationAndPush(5)
         }
       })
 
       setHandler(inFixedPoints, new InHandler {
         override def onPush(): Unit = {
-          log.info(s"inFixedPoints onPush() - setting fixedPoints")
-          fixedPoints = Option(grab(inFixedPoints))
-          runSimulationAndPush()
+          grabAllAvailable()
+          runSimulationAndPush(5)
         }
       })
 
       setHandler(inMovements, new InHandler {
         override def onPush(): Unit = {
-          log.info(s"inMovements onPush() - setting movements")
-          movements = Option(grab(inMovements))
-          runSimulationAndPush()
+          grabAllAvailable()
+          runSimulationAndPush(5)
         }
       })
 
-      def pushAndPull() = {
-        crunchStateWithSimulation match {
-          case None =>
-            log.info(s"Nothing to push")
-          case Some(cs) =>
-            log.info(s"Pushing CrunchStateWithSimulation")
-            push(outCrunch, cs)
-            crunchStateWithSimulation = None
-        }
+      setHandler(outCrunch, new OutHandler {
+        override def onPull(): Unit = pushAndPull()
+      })
 
-        allInlets.foreach(inlet => if (!hasBeenPulled(inlet)) pull(inlet))
+      def grabAllAvailable(): Unit = {
+        if (isAvailable(inShifts)) {
+          log.info(s"Grabbing available inShifts")
+          shiftsOption = Option(grab(inShifts))
+        }
+        if (isAvailable(inFixedPoints)) {
+          log.info(s"Grabbing available inFixedPoints")
+          fixedPointsOption = Option(grab(inFixedPoints))
+        }
+        if (isAvailable(inMovements)) {
+          log.info(s"Grabbing available inMovements")
+          movementsOption = Option(grab(inMovements))
+        }
+        if (isAvailable(inCrunch)) {
+          log.info(s"Grabbing available inCrunch")
+          crunchStateOption = Option(grab(inCrunch))
+        }
       }
 
-      def runSimulationAndPush() = {
-        log.info(s"Running simulation")
+      def runSimulationAndPush(eGateBankSize: Int): Unit = {
+        log.info(s"Running simulations")
 
         crunchStateWithSimulation = crunchStateOption.map {
-          case cs@ CrunchState(_, _, _, crunchMinutes) =>
+          case cs@CrunchState(_, _, _, crunchMinutes) =>
             val crunchMinutesWithDeployments = addDeployments(crunchMinutes, queueRecsToDeployments(_.toInt), staffDeploymentsByTerminalAndQueue, minMaxDesks)
             val crunchMinutesWithSimulation = crunchMinutesWithDeployments.groupBy(_.terminalName).flatMap {
               case (_, tcms) =>
                 tcms.groupBy(_.queueName).flatMap {
                   case (qn, qcms) =>
                     val minWlSd = qcms.toSeq.map(cm => Tuple3(cm.minute, cm.workLoad, cm.deployedDesks)).sortBy(_._1)
-                    val workLoads = minWlSd.map { case (_, wl, _) => wl }.toList
+                    val workLoads = minWlSd.map {
```

Comment:
might be worth extracting this into a function for clarity. applyEGateAdjustments or some such.